scripts/interfaces.ts

```diff
@@ -0,0 +1,167 @@
+// Native
+import { readFile, writeFile } from "fs/promises";
+import { mkdirSync } from "fs";
+import { basename, parse, dirname } from "path";
+
+// Vendor
+import glob from "glob";
+
+// Utilities
+import { createHashFromFile } from "./utilities/createFileHash";
+import { logger } from "./utilities/logger";
+import { spawnProcess } from "./utilities/spawnProcess";
+
+// TODO: investigate if there are ways we could generate these from compiled artifacts far more quickly
+// TODO: currently the generator stumbles on subsequent calls whenever an invalid ABI is generated (i.e. FusePoolDirectoryArbitrum.sol:18:51)
+// TODO: replace so we can parallelize the execution with limitations
```

Comment:
I feel like the PR is currently in a state where it makes sense to merge. The TODO's aren't critical except the second point but incompatible interfaces or ABI's can be filtered out easily when we spot them.

The IGNORE_LIST also makes it very explicit which files are not working correctly yet